GRSIProof/ExampleEventSelector.C

```diff
@@ -4,24 +4,25 @@
 
 #include "TH1.h"
 
-//need to define these histograms either globally or as members of the class
-TH1D* gE;
-TH1D* gE_b;
+void ExampleEventSelector::CreateHistograms() {
+   fH1["gE"] = new TH1D("gE","#gamma Singles",12000,0,3000);
+   fH1["gE_b"] = new TH1D("gE_b","#gamma Singles in rough #beta coincidence",12000,0,3000);
 
-void ExampleEventSelector::CreateHistograms()
-{
-   gE = new TH1D("gE","#gamma Singles",12000,0,3000);
-   gE_b = new TH1D("gE_b","#gamma Singles in rough #beta coincidence",12000,0,3000);
-   GetOutputList()->AddAll(gDirectory->GetList());
+	for(auto it : fH1) {
+		GetOutputList()->Add(it.second);
+	}
+	for(auto it : fH2) {
+		GetOutputList()->Add(it.second);
+	}
+	for(auto it : fHSparse) {
+		GetOutputList()->Add(it.second);
+	}
 }
 
-void ExampleEventSelector::FillHistograms()
-{
+void ExampleEventSelector::FillHistograms() {
    for(auto i = 0; i < fGrif->GetMultiplicity(); ++i){
-      gE->Fill(fGrif->GetGriffinHit(i)->GetEnergy());
+      fH1["gE"]->Fill(fGrif->GetGriffinHit(i)->GetEnergy());
```

Comment:
No, I don't think it's an issue. I've tested this on some data and run times before the change were 38.0, 34.6, and 37.6 seconds, after the change they were 35.2, 36.2, and 35.2 s.